src/Authentication/OAuth2Adapter.php

```diff
@@ -145,29 +145,42 @@ public function authenticate(Request $request, Response $response, MvcAuthEvent
             $request->getHeaders()->toArray()
         );
 
-        // Failure to validate
-        if (! $this->oauth2Server->verifyResourceRequest($oauth2request)) {
-            $oauth2Response = $this->oauth2Server->getResponse();
-            $status = $oauth2Response->getStatusCode();
-
-            // 401 or 403 mean invalid credentials or unauthorized scopes; report those.
-            if (in_array($status, [401, 403], true) && null !== $oauth2Response->getParameter('error')) {
-                return $this->mergeOAuth2Response($status, $response, $oauth2Response);
-            }
+        $token = $this->oauth2Server->getAccessTokenData($oauth2request);
```

Comment:
For anybody who comes to this issue wondering why `getAccessTokenData()` can be used in place of `verifyResourceRequest()`: `OAuth2\Server::verifyResourceRequest()` proxies to `OAuth2\Controller\ResourceController::verifyResourceRequest()`. Internally, it performs `getAccessTokenData()`, and, if a token is present, returns `true`. It will _optionally_ do a scope verification before doing so, but only if a `$scope` argument is provided — and we do not provide that. As such, `getAccessTokenData()` returns `null` if the token is invalid in any way, and the token otherwise, making it sufficient for our needs.